Spigot-API-Patches/0244-added-PlayerTradeEvent.patch

```diff
@@ -130,4 +121,54 @@ index 0000000000000000000000000000000000000000..198e5464eae6b961c83148a57c18f91a
 +    public static HandlerList getHandlerList() {
 +        return handlers;
 +    }
++
++}
+diff --git a/src/main/java/io/papermc/paper/event/player/PlayerTradeEvent.java b/src/main/java/io/papermc/paper/event/player/PlayerTradeEvent.java
+new file mode 100755
+index 0000000000000000000000000000000000000000..a83a7f6c4f528b820320ba33c8d8ab074d2b2051
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/player/PlayerTradeEvent.java
+@@ -0,0 +1,43 @@
++package io.papermc.paper.event.player;
++
++import org.bukkit.entity.AbstractVillager;
++import org.bukkit.entity.Player;
++import org.bukkit.event.HandlerList;
++import org.bukkit.inventory.MerchantRecipe;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when a player trades with a villager or wandering trader
++ */
++public class PlayerTradeEvent extends PlayerPurchaseEvent {
++
++    private static final HandlerList handlers = new HandlerList();
```

Comment:
with a HandlerList here, PlayerPurchaseEvent listeners won't be called when PlayerTradeEvent is fired, not sure if this is intended or not but I doubt it

Reply:
Yeah, I have no idea how to deal with that.. would it even work? Having the `HandlerList` on `PlayerPurchaseEvent`? Would `PlayerTradeEvent` be unaffected by that?

Reply:
if it's on purchase event, listeners of trade and purchase will be called when trade is fired, and only purchase when purchase is fired

Reply:
So, removing the `HandlerList` from `PlayerTradeEvent` would mean:

```java
@EventHandler
public void onTradeEvent(PlayerTradeEvent event) {
    // This event will receive only PlayerTradeEvent
}

@EventHandler
public void onPurchaseEvent(PlayerPurchaseEvent event) {
    // This event will receive PlayerTradeEvent and PlayerPurchaseEvent
}
```

??

Reply:
yes

Reply:
Fair enough, updating now